Minecraft Purpur version download 404 error

### Enhancement Type

Improve an existing feature

### Describe the enhancement

Hi, the Purpur version is giving the following error when downloading

```
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 2 1000 1000 4096 Jun 13 19:02 /data'
[init] Resolved version given LATEST into 1.20.1 and major version 1.20
[init] Resolving type given PURPUR
[mc-image-helper] 20:24:00.596 ERROR : 'get' command failed. Version is 1.30.1
me.itzg.helpers.get.RequestFailedException: Failed to download https://api.purpurmc.org/v2/purpur/1.20.1: status code: 404, reason phrase: Not Found
	at me.itzg.helpers.get.GetCommand.processSingleUri(GetCommand.java:442)
	at me.itzg.helpers.get.GetCommand.call(GetCommand.java:188)
	at me.itzg.helpers.get.GetCommand.call(GetCommand.java:55)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:145)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:145)
Caused by: org.apache.hc.client5.http.HttpResponseException: status code: 404, reason phrase: Not Found
	at org.apache.hc.client5.http.impl.classic.AbstractHttpClientResponseHandler.handleResponse(AbstractHttpClientResponseHandler.java:68)
	at me.itzg.helpers.http.LoggingResponseHandler.handleResponse(LoggingResponseHandler.java:20)
	at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:247)
	at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188)
	at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:162)
	at me.itzg.helpers.get.GetCommand.processSingleUri(GetCommand.java:440)
	... 11 common frames omitted
[init] ERROR: Failed to locate a Purpur build for 1.20.1.
[init]        Please check if a download is available at https://purpur.pl3x.net/downloads/
```

My docker compose file:

```
version: "3"

services:
  mc:
    image: itzg/minecraft-server
    container_name: "minecraftJS"
    ports:
      - 25565:25565
    environment:
      USE_AIKAR_FLAGS: true
      EULA: "TRUE"
      VERSION: LATEST
      TYPE: PURPUR
      ONLINE_MODE: false
      MEMORY: 2G
    tty: true
    stdin_open: true
    restart: unless-stopped
    volumes:
      # attach a directory relative to the directory containing this compose file
      - ./minecraft-data:/data
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -46,7 +46,7 @@ RUN easy-add --var os=${TARGETOS} --var arch=${TARGETARCH}${TARGETVARIANT} \
   --var version=0.1.1 --var app=maven-metadata-release --file {{.app}} \
   --from https://github.com/itzg/{{.app}}/releases/download/{{.version}}/{{.app}}_{{.version}}_{{.os}}_{{.arch}}.tar.gz
 
-ARG MC_HELPER_VERSION=1.30.1
+ARG MC_HELPER_VERSION=1.31.0
 ARG MC_HELPER_BASE_URL=https://github.com/itzg/mc-image-helper/releases/download/${MC_HELPER_VERSION}
 # used for cache busting local copy of mc-image-helper
 ARG MC_HELPER_REV=1

--- a/scripts/start-deployPurpur
+++ b/scripts/start-deployPurpur
@@ -2,51 +2,47 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+: "${PURPUR_BUILD:=LATEST}"
 : "${PURPUR_DOWNLOAD_URL:=}"
 
 # shellcheck source=start-utils
 . "${SCRIPTS:-/}start-utils"
 isDebugging && set -x
 
+resultsFile=/data/.purpur.env
+
 if [[ $PURPUR_DOWNLOAD_URL ]]; then
-  export SERVER=$(getFilenameFromUrl "${PURPUR_DOWNLOAD_URL}")
-
-  if [ -f "$SERVER" ]; then
-    zarg=(-z "$SERVER")
-  fi
-
-  echo "Preparing custom Purpur jar from $PURPUR_DOWNLOAD_URL"
-
-  curl -fsSL -o "$SERVER" "${zarg[@]}" "${PURPUR_DOWNLOAD_URL}"
-else
-  : "${VANILLA_VERSION:?}"
-  : "${PURPUR_BUILD:=LATEST}"
-  : "${FORCE_REDOWNLOAD:=false}"
-
-  if [[ ${PURPUR_BUILD} == LATEST ]]; then
-    if ! PURPUR_BUILD=$(get --json-path=".builds.latest" "https://api.purpurmc.org/v2/purpur/${VANILLA_VERSION}"); then
-      log "ERROR: Failed to locate a Purpur build for ${VANILLA_VERSION}."
-      log "       Please check if a download is available at https://purpur.pl3x.net/downloads/"
+  if ! mc-image-helper install-purpur \
+    --output-directory=/data \
+    --results-file="$resultsFile" \
+    --url="$PURPUR_DOWNLOAD_URL}"; then
+      log "ERROR: failed to download from custom Purpur URL"
       exit 1
-    fi
   fi
+  # grab SERVER and export it
+  set -a
+  # shellcheck disable=SC1090
+  source "${resultsFile}"
+  set +a
 
-  export SERVER="purpur-${VANILLA_VERSION}-${PURPUR_BUILD}.jar"
-
-  log "Removing old Purpur versions ..."
-  shopt -s nullglob
-  for f in purpur-*.jar; do
-    [[ $f != "$SERVER" ]] && rm "$f"
-  done
-
-  if [ ! -f "$SERVER" ] || isTrue "$FORCE_REDOWNLOAD"; then
-      downloadUrl="https://api.purpurmc.org/v2/purpur/${VANILLA_VERSION}/${PURPUR_BUILD}/download"
-      log "Downloading Purpur from $downloadUrl ..."
-      if ! get -o "$SERVER" "$downloadUrl"; then
-        log "ERROR: failed to download from $downloadUrl (status=$?)"
-        exit 3
-      fi
+else
+  args=(
+    --output-directory=/data
+    --results-file="$resultsFile"
+    --version="$VERSION"
+  )
+  if [[ $PURPUR_BUILD ]]; then
+    args+=(--build="$PURPUR_BUILD")
   fi
+  if ! mc-image-helper install-purpur "${args[@]}"; then
+      log "ERROR: failed to download Purpur"
+      exit 1
+  fi
+  # grab SERVER and export it
+  set -a
+  # shellcheck disable=SC1090
+  source "${resultsFile}"
+  set +a
 fi
 
 # Normalize on Spigot for later operations

